fplll/enum/enumerate.h

```diff
@@ -44,7 +45,16 @@ template <typename ZT, typename FT> class EnumerationDyn : public EnumerationBas
                  const vector<enumf> &pruning = vector<enumf>(), bool dual = false,
                  bool subtree_reset = false);
 
-  inline uint64_t get_nodes() const { return nodes; }
+  inline uint64_t get_nodes(const int level = -1) const
+  {
+    if (level == -1)
+    {
+      return std::accumulate(nodes.cbegin(), nodes.cend(), 0);
+    }
+    return nodes[level];
```

Comment:
Do we just want to accept this buffer overread or shall we do a bounds check?

Reply:
So I didn't do this for the following reasons:

(a) the .at() method provides the checks you'd want, but it throws an
exception: I dont know if we want to enforce the handling that on the
calling code.

(b) the array is quite big, so in the cases we over-read it should return a
0: which seems like the most sensible return value for this.

(c) if we did the checks for this, what would we return? We could return
~uint64_t(0) for invalid, but then we would have to check for that return
type in all places (which would likely screw up the count by quite a bit).


What we could do instead is this: we can use an assert for this (level =<
FPLLL_MAX_ENUM_DIM). That way there won't be a compile time cost if DNDEBUG
is enabled, but for debugging it would flag any potential errors.

Let me know what you think.

Reply:
I guess we don't necessarily enforce bounds elsewhere either,e.g. when giving enumeration dimensions, so you're right, we might as well leave it here and push that responsibility on FPyLLL and friends.